src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -31,63 +73,153 @@ public ConsoleLoggerProcessor(IConsole console, IConsole errorConsole)
                 Name = "Console logger queue processing thread"
             };
             _outputThread.Start();
+            _processorId = ++s_processorCount;
+            s_processors.Add(this);
         }
 
-        public virtual void EnqueueMessage(LogMessageEntry message)
+        static ConsoleLoggerProcessor()
         {
-            if (!_messageQueue.IsAddingCompleted)
-            {
-                try
-                {
-                    _messageQueue.Add(message);
-                    return;
-                }
-                catch (InvalidOperationException) { }
-            }
+            s_meter.CreateObservableGauge<long>("queue-size", GetQueueSize);
+        }
 
-            // Adding is completed so just log the message
-            try
+        public virtual void EnqueueMessage(LogMessageEntry message)
```

Comment:
Does this need to be `virtual`? Is it only for testing?

Reply:
yes I believe it is used for testing